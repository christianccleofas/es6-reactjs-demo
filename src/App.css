.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.body-wrapper {
  padding-left: 5%;
}

.app-bar {
  background-color: #006241 !important;
  position: fixed;
}

.order-item-name, .order-item-subTotal {
  font-size: 1em; 
  font-weight: 600;
  color: #23423c;
  letter-spacing: -1;
}

.MuiPaper-root.MuiDrawer-paper.MuiDrawer-paperAnchorRight.MuiPaper-elevation16 {
  width: 20%;
}