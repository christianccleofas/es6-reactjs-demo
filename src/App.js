import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import "./App.css";
import {run} from './utils/main'
import Item from './components/common/Item/Item';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export default function ButtonAppBar() {
  const classes = useStyles();
  run();
  return (
    <div className={classes.root} id="root">
      <AppBar position="static" className="app-bar">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton>
          <Typography variant="h5" className={classes.title}>
            ES6 + React JS Demo
          </Typography>
          <Button color="inherit"></Button>
        </Toolbar>
      </AppBar>
      <div className="body-wrapper">
        <Item />
      </div>
    </div>
  );
}