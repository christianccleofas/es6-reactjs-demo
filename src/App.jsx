import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import "./App.css";
import _ from 'underscore';
import CartBadge from './CartBadge/CartBadge';
import CartDrawer from './CartDrawer/CartDrawer';
import Menu from './components/container/Menu/Menu';
import SnackbarFeedback from './SnackbarFeedback/SnackbarFeedback';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export default function ButtonAppBar() {
  const classes = useStyles();
  const [orderList, setOrderList] = React.useState([]);
  const [isCartDrawerOpen, setIsCartDrawerOpen] = React.useState(false);
  const [isSnackbarFeedbackOpen, setIsSnackbarFeedbackOpen] = React.useState(false);

  const openCartDrawer = () => setIsCartDrawerOpen(true); //opens cart drawer
  const closeCartDrawer = () => setIsCartDrawerOpen(false); //closes cart drawer

  const getOrderFromMenu = orderItem => {
    /* Adds item to the list of orders on the cart drawer when Add to Cart button is clicked */
    if(!orderList.length) {
      /* no orders yet */
      setOrderList([...orderList, orderItem]);
    } else {
      /* orders */
      let isItemExisting = _.findWhere(orderList, {name: orderItem.name});
      let itemIndex = _.indexOf(orderList, isItemExisting);
      let orders = orderList;
      /* no duplicate orders should be displayed */
      if(itemIndex >= 0) {
        orders[itemIndex]['qty'] = orderItem.qty;
        setOrderList(orders);
      } else {
        setOrderList([...orderList, orderItem]);
      }
    }
    setIsSnackbarFeedbackOpen(true);
  }

  const removeItemFromCart = data => {
    /* Removes item from the list of orders to the cart drawer when removed button is clicked */
    const listOfItemsAfterRemoval = _.without(orderList, data);
    setOrderList(listOfItemsAfterRemoval);
  }

  const customProps = {
    orderList,
    openCartDrawer,
    closeCartDrawer,
    isCartDrawerOpen,
    getOrderFromMenu,
    removeItemFromCart
  }

  const handleClose = (event, reason) => {
    /* Close snackbar component */
    if (reason === 'clickaway') {
      return;
    }

    setIsSnackbarFeedbackOpen(false);
  };


  const customSnackbarProps = {
    /* Snackbar props */
    handleClose,
    openCartDrawer,
    isSnackbarFeedbackOpen,
  }

  return (
    <div className={classes.root} id="root">
      <AppBar position="static" className="app-bar">
        <Toolbar>
          <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon/>
          </IconButton>
          <Typography variant="h6" className={classes.title}>
              ES6 | React JS Demo
          </Typography>
          <CartBadge {...customProps} />
        </Toolbar>
      </AppBar>
      <CartDrawer {...customProps} />
      <div className="body-wrapper">
        <Menu {...customProps} />
      </div>
      <SnackbarFeedback {...customSnackbarProps} />
    </div>
  )
}