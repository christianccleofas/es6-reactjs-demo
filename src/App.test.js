import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import toJson from 'enzyme-to-json';

test('renders learn react link', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
