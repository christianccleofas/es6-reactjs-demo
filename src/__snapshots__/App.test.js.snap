// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="makeStyles-root-1"
  id="root"
>
  <WithStyles(ForwardRef(AppBar))
    className="app-bar"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className="makeStyles-menuButton-2"
        color="inherit"
        edge="start"
      >
        <Memo(ForwardRef(MenuIcon)) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h5"
      >
        ES6 + React JS Demo
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        color="inherit"
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <div
    className="body-wrapper"
  >
    <Item />
  </div>
</div>
`;
